packages/shared/src/modules/users/libs/validation-schemas/user-sign-up.validation-schema.ts

```diff
@@ -11,14 +11,29 @@ const userSignUp: z.ZodType<UserSignUpRequestDto> = z
 			.min(UserValidationRule.EMAIL_MINIMUM_LENGTH, {
 				message: UserValidationMessage.EMAIL_REQUIRED,
 			})
-			.email({
-				message: UserValidationMessage.EMAIL_WRONG,
-			}),
+			.max(UserValidationRule.EMAIL_MAXIMUM_LENGTH, {
+				message: UserValidationMessage.EMAIL_TOO_LONG,
+			})
+			.refine(
+				(email) => {
+					const [localPart, domainPart] = email.split("@");
+
+					return (
+						!!localPart &&
+						!!domainPart &&
+						UserValidationRule.EMAIL_LOCAL_PART_PATTERN.test(localPart) &&
+						UserValidationRule.EMAIL_DOMAIN_PART_PATTERN.test(domainPart)
+					);
+				},
+				{
+					message: UserValidationMessage.EMAIL_WRONG,
+				},
+			),
```

Comment:
Doesn't Zod have built-in validation for emails? I think this logic can be fully covered by it. Also, let's add email validation for sign-in too (for password is not needed, users should know their passwords).

Reply:
Yes, but this validation break some [rules](https://github.com/BinaryStudioAcademy/bsa-2024-gitfit/issues/131):

- a hyphen should not be placed next to a dot;
- a hyphen should not be at the beginning or end of the local part;

P.S. I thought [it](https://github.com/BinaryStudioAcademy/bsa-2024-gitfit/issues/131#issuecomment-2321043852) shouldn't pass, but now I'm not sure (about Local Part)

Reply:
I think for our project such a strict email validation is not needed. So it's better to keep built-in email validation and ask qa to adjust the rules.

Reply:
Please agree this with qa first before changing

Reply:
agreed on simplifying :+1: